packages/ui/src/components/va-modal/VaModal.vue

```diff
@@ -213,7 +214,9 @@ export default defineComponent({
     const onAfterLeaveTransition = (el: HTMLElement) => emit('close', el)
 
     const listenKeyUp = (e: KeyboardEvent) => {
-      if (e.code === 'Escape' && !props.noEscDismiss && !props.noDismiss) {
+      const isLastNestedModal = !modal.value?.$el.nextElementSibling
```

Comment:
in the `body` after the last modal there can be one or more dropdowns (as well as user elements)

Reply:
I wonder If this could actually be a feature? So dropdown get dismissed on esc before the popup.

Reply:
Components like `VaToast` do not respond (are not removed from the `body`) when the `esc` key is pressed.